template/core/data/src/main/java/io/bloco/core/data/network/OpenLibraryService.kt

```diff
@@ -1,34 +1,40 @@
 package io.bloco.core.data.network
 
-import io.bloco.core.data.models.Book
+import io.bloco.core.data.models.BookDetailsDto
 import io.bloco.core.data.models.BookRecords
-import io.ktor.client.HttpClient
 import io.ktor.client.call.body
 import io.ktor.client.request.get
 import io.ktor.client.request.url
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 class OpenLibraryService
 @Inject constructor(
-    @OpenLibraryHttpClient private var httpClient: HttpClient
+    private val httpClient: OpenLibraryHttpClient
 ) {
-    suspend fun getBooks(): Result<BookRecords> = try {
-        Result.success(
-            httpClient.get {
-                url(path = "/")
-            }.body()
-        )
-    } catch (e: Exception) {
-        Result.failure(e)
+
+    suspend fun getBooks(): Result<BookRecords> = withContext(Dispatchers.IO) {
```

Comment:
I believe ktor already handles the network calls in a background dispatcher, so we don't need to change dispatchers here. Also, never hardcode a scope/context/dispatcher, it always should be swappable, or at least exposed.

Reply:
strict mode was crashing, hence the change. 
But yes I shall make it a argument

Reply:
Is there a chance we're missing some ktor config then?

Reply:
going to check